### PR TITLE
feat: add support infrastructure

### PR DESCRIPTION
**Pull request by josefaidt**
*Issue #, if available:*

#135

*Description of changes:*

- adds ec2 instance
	- with subdomain support
	- access to filesystem for sqlite backup
- adds [litestream](https://litestream.io) sidecar container for SQLite backups
- adds s3 bucket for database backups
- consolidates logical ID casing
- migrates database files to easier location, `/data/*.db`
- updates architecture diagram

image

*TODO:*

- [x] ec2 instance to have access to prisma schema, migrations
	- mitigated by deciding to temporarily clone repo when we need to run migrations
- [x] ~~install and run [litestream](https://litestream.io/) on instance start~~
	- mitigated by adding sidecar container to backup to s3
- [x] ensure backups are uploaded to s3
- [x] add architecture diagram

*Migration Notes:*

This PR introduces a new DATABASE_URL file path. Move your existing database to the new location with the following:

```bash
mv db prisma/data
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cdk/src/stack.ts [hunk 1]
```diff
@@ -45,42 +47,57 @@ export class HeyAmplifyStack extends Stack {
...
         },
       },
       secrets,
-      subdomain: props.subdomain
-        ? new AmplifyAwsSubdomain(this, 'subdomain', props.subdomain)
-        : undefined,
+      subdomain,
       filesystem,
-      filesystemMountPoint: '/usr/src/db',
+      filesystemMountPoint: '/data',
```
Comment (dnys1) [inline cdk/src/stack.ts:93]: Looks like `filesystemMountPoint` and `DATABASE_URL` could potentially get out of sync which would affect the litestream container which uses both vars. Should they be related somehow?

File: scripts/db/litestream.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+access-key-id: ${AWS_ACCESS_KEY_ID}
+secret-access-key: ${AWS_SECRET_ACCESS_KEY}
+
+dbs:
+  - path: /data
```
Comment (dnys1) [inline scripts/db/litestream.yml:5]: Should this be an env var as well?

File: scripts/db/litestream.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    replicas:
+      - url: s3://${BUCKET}/backup
```
Comment (dnys1) [inline scripts/db/litestream.yml:7]: This differs from the container definition (`backup` vs  `backups`). Does that matter at all?

**Review by dnys1 — APPROVED** [review]

File: scripts/db/litestream.yml (hunk 3 above)
Comment (josefaidt) [reply scripts/db/litestream.yml]: see above, file has been removed

File: cdk/src/stack.ts (hunk 1 above)
Comment (josefaidt) [reply cdk/src/stack.ts]: Great callout. I've refactored this to set `filesystemMountPoint` outside of this config and construct the `DATABASE_URL` env var using this variable

```ts
DATABASE_URL: `file:${filesystemMountPoint}/${this.envName}.db`,
```

**Pull request merged**